File: Java/Platform/Basics/AsyncParallelTest/Program1.java
import java.util.stream.IntStream;

class Program {

	public static void main(String[] args) throws Exception {
		int limit = Integer.parseInt(args[0]);
		System.out.print("Computing...");
		var c = new Computation();
		var t1 = System.nanoTime();
		var r = c.compute(1, limit);
		var t2 = System.nanoTime();
		System.out.println("Done!");
		System.out.printf("Result = %d, computed in %.3f seconds.%n", r, (t2 - t1) / 1e9);
	}

	static class Computation {

		public long compute(int first, int last) {
			return IntStream.range(first, last + 1)
					.mapToLong(Worker::doWork)
					.sum();
		}
	}
}

File: dotnet/Platform/Basics/AsyncParallelTest/DemoApp/Program1.cs
class Program
{
    private static void DoComputation(int limit)
    {
        Console.Write("Computing...");
        var c = new Computation();
        var w = new System.Diagnostics.Stopwatch();
        w.Start();
        long r = c.Compute(1, limit);
        w.Stop();
        Console.WriteLine("Done!");
        Console.WriteLine("Result = {0}, computed in {1:0.000} seconds.", r, w.Elapsed.TotalSeconds);
    }

    static void Main(string[] args)
    {
        int n = int.Parse(args[0]);
        DoComputation(n);
    }

    class Computation
    {
        public long Compute(int first, int last)
        {
            return Enumerable.Range(first, last - first + 1)
                        .Select(Worker.DoWork)
                        .Sum();
        }
    }
}